lib/dry/types.rb

```diff
@@ -27,29 +27,55 @@ module Types
     extend Dry::Core::ClassAttributes
     include Dry::Core::Constants
 
-    # @!attribute [r] namespace
-    #   @return [Container{String => Definition}]
-    defines :namespace
-
-    namespace self
-
     TYPE_SPEC_REGEX = %r[(.+)<(.+)>].freeze
 
-    # @return [Module]
-    def self.module
-      namespace = Module.new
-      define_constants(namespace, type_keys)
-      namespace.extend(BuilderMethods)
-      namespace
-    end
-
-    # @deprecated Include {Dry::Types.module} instead
-    def self.finalize
-      warn 'Dry::Types.finalize and configuring namespace is deprecated. Just'\
-       ' do `include Dry::Types.module` in places where you want to have access'\
-       ' to built-in types'
-
-      define_constants(self.namespace, type_keys)
+    # Export registered types as a module with constants
+    #
+    # @example no options
+    #
+    #   module Types
+    #     # exports all types as constants, uses modules for namespaces
+    #     include Dry::Types.module
+    #   end
+    #   # nominal types are exported by default
+    #   Types::Integer
+    #   # => #<Dry::Types[Definition<Integer>]>
+    #   Types::Strict::Integer
+    #   # => #<Dry::Types[Constrained<Definition<Integer> rule=[type?(Integer)]>]>
+    #
+    # @example changing default types
+    #
+    #   module Types
+    #     include Dry::Types(default: :strict)
```

Comment:
There's no `Dry.Types` method, do you intend to add it eventually?

Reply:
yeah, will do it today and merge the whole thing shortly after that